src/input_common/helpers/joycon_protocol.cpp

```diff
@@ -0,0 +1,362 @@
+        imu.gyro[i].scale = static_cast<s16>(buffer[index + 18] | (buffer[index + 19] << 8));
+    }
+    return imu;
+}
+
+CalibrationData GetUserCalibrationData(JoyconHandle& joycon_handle,
+                                       ControllerType controller_type) {
+    CalibrationData calibration{};
+    switch (controller_type) {
+    case ControllerType::Left:
+        calibration.left_stick = GetLeftJoyStickCalibration(joycon_handle, false);
+        break;
+    case ControllerType::Right:
+        calibration.right_stick = GetRightJoyStickCalibration(joycon_handle, false);
+        break;
+    case Joycon::ControllerType::Pro:
+        calibration.left_stick = GetLeftJoyStickCalibration(joycon_handle, false);
+        calibration.right_stick = GetRightJoyStickCalibration(joycon_handle, false);
+        break;
+    default:
+        break;
+    }
+
+    calibration.imu = GetImuCalibration(joycon_handle, false);
+    return calibration;
+}
+
+CalibrationData GetFactoryCalibrationData(JoyconHandle& joycon_handle,
+                                          ControllerType controller_type) {
+    CalibrationData calibration{};
+    switch (controller_type) {
+    case ControllerType::Left:
+        calibration.left_stick = GetLeftJoyStickCalibration(joycon_handle, true);
+        break;
+    case ControllerType::Right:
+        calibration.right_stick = GetRightJoyStickCalibration(joycon_handle, true);
+        break;
+    case Joycon::ControllerType::Pro:
+        calibration.left_stick = GetLeftJoyStickCalibration(joycon_handle, true);
+        calibration.right_stick = GetRightJoyStickCalibration(joycon_handle, true);
+        break;
+    default:
+        break;
+    }
+
+    calibration.imu = GetImuCalibration(joycon_handle, true);
+    return calibration;
+}
+
+void EnableImu(JoyconHandle& joycon_handle, bool enable) {
+    SDL_hid_set_nonblocking(joycon_handle.handle, 0);
+    const std::vector<u8> buffer{static_cast<u8>(enable ? 1 : 0)};
+    SendSubCommand(joycon_handle, SubCommand::ENABLE_IMU, buffer, buffer.size());
+    SDL_hid_set_nonblocking(joycon_handle.handle, 1);
+}
+
+void EnableRumble(JoyconHandle& joycon_handle, bool enable) {
+    SDL_hid_set_nonblocking(joycon_handle.handle, 0);
+    const std::vector<u8> buffer{static_cast<u8>(enable ? 1 : 0)};
+    SendSubCommand(joycon_handle, SubCommand::ENABLE_VIBRATION, buffer, buffer.size());
+    SDL_hid_set_nonblocking(joycon_handle.handle, 1);
+}
+
+void SendData(JoyconHandle& joycon_handle, std::span<const u8> buffer, std::size_t size) {
+    SDL_hid_write(joycon_handle.handle, buffer.data(), size);
+}
+
+std::vector<u8> GetResponse(JoyconHandle& joycon_handle, SubCommand sc) {
+    constexpr int timeout_mili = 100;
+    int tries = 0;
+    std::vector<u8> buffer(max_resp_size);
+    do {
+        int result =
+            SDL_hid_read_timeout(joycon_handle.handle, buffer.data(), max_resp_size, timeout_time);
```

Comment:
```suggestion
            SDL_hid_read_timeout(joycon_handle.handle, buffer.data(), max_resp_size, timeout_mili);
```

I'm sorry, but do you actually verify that it builds before pushing?